src/main/java/org/magicdgs/thaplv/cmd/argumentcollections/HaplotypeModelArgumentCollection.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2016, Daniel Gomez-Sanchez <daniel.gomez.sanchez@hotmail All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are permitted
+ * provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of conditions
+ * and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions
+ * and the following disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse
+ * or promote products derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package org.magicdgs.thaplv.cmd.argumentcollections;
+
+import org.magicdgs.thaplv.cmd.ThaplvArgumentDefinitions;
+import org.magicdgs.thaplv.haplotypes.model.HaplotypeModel;
+import org.magicdgs.thaplv.haplotypes.model.VariantHaplotypeConverter;
+
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.ArgumentCollectionDefinition;
+import org.broadinstitute.hellbender.exceptions.UserException;
+
+/**
+ * Argument collection for haplotype models
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public abstract class HaplotypeModelArgumentCollection implements ArgumentCollectionDefinition {
+    private static final long serialVersionUID = 1L;
+
+    @Argument(fullName = ThaplvArgumentDefinitions.HAPLOTYPE_MODEL_LONG, shortName = ThaplvArgumentDefinitions.HAPLOTYPE_MODEL_SHORT, doc = "Haplotype model for convert calls.", common = false, optional = true)
+    public HaplotypeModel haplotypeModel = HaplotypeModel.HAPLOID;
+
+    /**
+     * Get the haplotype converter for this tool
+     * @param allowDontCheck is {@link HaplotypeModel#DONT_CHECK} model allowed?
+     */
+    public abstract VariantHaplotypeConverter getHaplotypeConverter(final boolean allowDontCheck);
```

Comment:
Make protected, add constructor for `allowDontCheck` and a public method without parameters. Then remove the `getHaplotypeConverter()` to don't expose this to implemented `HaplotypeWalker`. We don't want to override this method anyway.